src/app/newsFeedPage/page.tsx

```diff
@@ -1,19 +1,65 @@
 'use client';
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { IoIosArrowRoundBack } from 'react-icons/io';
 import NavBar from '../../components/userComponents/navBar/navBar';
+import { NewsRow } from '../../types/types';
+import { fetchAllNewsByDate } from '../../supabase/news/queries';
+import NewsDisplay from '../../components/userComponents/NewsDisplay/NewsDisplay';
 
 /**
- * @returns The news feed page
+ * @param evt on click of button
+ */
+function goBack(evt: React.SyntheticEvent) {
+  // ignore the native anchor action
+  evt.preventDefault();
+
+  window.history.back();
+}
+
+/**
+ * @returns back button
+ */
+function BackButton() {
+  return (
+    <button type="button" onClick={goBack} className="text-scary-forest">
+      {' '}
+      <IoIosArrowRoundBack size={40} />
+    </button>
+  );
+}
+
+/**
+ * @returns news feed page
  */
 function App() {
```

Comment:
pretty similar here to [nextjs's new data fetching scheme](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating), might want to consider using it.

```
async function getData() {
  const res = await fetch('https://api.example.com/...')
  // The return value is *not* serialized
  // You can return Date, Map, Set, etc.
 
  if (!res.ok) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data')
  }
 
  return res.json()
}
 
export default async function Page() {
  const data = await getData()
 
  return <main></main>
}
```